### PR TITLE
refactor(prediction): Convert Global State to Local State and add Config Provider

### PR DESCRIPTION
**Pull request by chefilip**
Purpose: Enable to have multiple prediction token version in the future.
Tasks: 
- Convert Prediction Global State into Local State
- Generalize the Prediction Config

### CODE REVIEW

File: src/pages/prediction/leaderboard.tsx [hunk 1]
```diff
@@ -1,3 +1,18 @@
...
+
 import PredictionsLeaderboard from '../../views/Predictions/Leaderboard'
 
-export default PredictionsLeaderboard
+const store = makeStore(reducers, initialState, config)
+
+export default () => {
+  return (
+    <ConfigProvider>
+      <LocalReduxProvider store={store}>
```
Comment (chef-jojo) [inline src/pages/prediction/leaderboard.tsx:13]: we don't need to make different store for prediction and leaderboard page, right?

File: src/state/predictions/hooks.ts [hunk 2]
```diff
@@ -136,6 +138,31 @@ export const useGetOrFetchLeaderboardAddressResult = (account: string) => {
...
 
-export const useGetSelectedAddress = () => {
-  return useSelector((state: State) => state.predictions.leaderboard.selectedAddress)
+export const useGetSelectedAddress = (): string => {
+  return useSelector((state: PredictionsState) => state.leaderboard.selectedAddress)
+}
+
+// Because Modal Component is rendered outside the Prediction Page contexts
+// We have to pass local state as props instead of retrieving directly in component
+export const useStatModalProps = (account?: string) => {
```
Comment (chef-jojo) [inline src/state/predictions/hooks.ts:147]: umm can we inject the provider in _app.tsx?

File: src/pages/prediction/leaderboard.tsx (hunk 1 above)
Comment (chefilip) [reply src/pages/prediction/leaderboard.tsx]: Good catch.

File: src/state/predictions/hooks.ts (hunk 2 above)
Comment (chefilip) [reply src/state/predictions/hooks.ts:147]: we couldn't achieve various prediction config if Injecting Config Provider in `_app.tsx`. The flaw is in the Modal component design. It should use Portal to keep the Context.

**Review by chef-jojo — APPROVED** [review]

**Pull request merged**